pkg/controller/plan/kubevirt.go

```diff
@@ -503,12 +523,69 @@ func (r *KubeVirt) EnsureDataVolumes(vm *plan.VMStatus, dataVolumes []cdi.DataVo
 					dv.Name),
 				"vm",
 				vm.String())
+			vm.VolumeNames = append(vm.VolumeNames, dv.Name)
 		}
 	}
 
 	return
 }
 
+//
+// Return DataVolumes associated with a VM.
+func (r *KubeVirt) getDVs(vm *plan.VMStatus) (dvs []DataVolume, err error) {
+	dvsList := &cdi.DataVolumeList{}
+	err = r.Destination.Client.List(
+		context.TODO(),
+		dvsList,
+		&client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(r.vmLabels(vm.Ref)),
+			Namespace:     r.Plan.Spec.TargetNamespace,
+		})
+
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	dvs = []DataVolume{}
+	for i := range dvsList.Items {
+		dv := &dvsList.Items[i]
+		dvs = append(dvs, DataVolume{
+			DataVolume: dv,
+			PVC:        nil,
+		})
+	}
+	return
+}
+
+//
+// Return PersistentVolumeClaims associated with a VM.
+func (r *KubeVirt) getPVCs(vm *plan.VMStatus) (pvcs []core.PersistentVolumeClaim, err error) {
+	pvcsList := &core.PersistentVolumeClaimList{}
+	err = r.Destination.Client.List(
+		context.TODO(),
+		pvcsList,
+		&client.ListOptions{
+			Namespace: r.Plan.Spec.TargetNamespace,
+		})
+
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	pvcs = []core.PersistentVolumeClaim{}
+	for i := range pvcsList.Items {
+		pvc := &pvcsList.Items[i]
+		for _, volumeName := range vm.VolumeNames {
+			if volumeName == pvc.GetName() {
+				pvcs = append(pvcs, *pvc)
+			}
```

Comment:
above we get the PVC in the namespace we target (pvcsList). but, we don't know which are used by our VM. since we have them now in `vm.VolumeNames` we can tell which one is relevant. the DV and PVC names are the same in the target namespace.

Reply:
ah ok, that makes sense. is it only used in the current flow? if so, how about adding a label that correlate the PVCs with the VM and then query with label selector?

Reply:
since the PVC are made by the DV, we will need anyway to pass and check the name. i don't see a real benefit in adding the labels, however it can easily be done in this for loop for future use (if needed).

Reply:
maybe I'm missing something but I see potential benefit in terms of implementation - if we can set labels on the PVCs, that are propagated from their DVs, so later we wouldn't query all PVCs in the namespace and compare them against the set of name that we store elsewhere but rather invoke a query that returns all PVCs that are to be associated with the VM, wouldn't it be a cleaner way to do that?